src/bdc/steps/helpers/generate_hash_leads.py

```diff
@@ -0,0 +1,80 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2023 Ruchita Nathani <ruchita.nathani@fau.de>
+
+import hashlib
+import os
+from datetime import datetime
+
+import pandas as pd
+
+from bdc.steps.step import Step
+from database import get_database
+from logger import get_logger
+
+log = get_logger()
+
+
+class LeadHashGenerator:
+    BASE_PATH = os.path.dirname(__file__)
+    _curr_lookup_table = ("", None)
+
+    def hash_lead(self, lead_data):
+        # Concatenate key lead information
+        data_to_hash = (
+            str(lead_data["First Name"])
+            + str(lead_data["Last Name"])
+            + str(lead_data["Company / Account"])
+            + str(lead_data["Phone"])
+            + str(lead_data["Email"])
+        )
+
+        # Hash the concatenated string using SHA-256
+        hash = hashlib.sha256(data_to_hash.encode()).hexdigest()
+
+        return hash
+
+    def hash_check(
```

Comment:
So basically after creating the hash, you'll check the lookup table if it has the data for this step and lead and return it instead of executing this step, correct?

Reply:
Exactly if the hash is in the lookup table and the dataframe contains data in the specified columns then that data is used instead of calculating it again